src/hivpy/hiv_testing.py

```diff
@@ -108,6 +108,42 @@ def update_hiv_testing(self, pop):
                 pop.set_present_variable(col.NP_LAST_TEST, 0,
                                          sub_pop=pop.apply_bool_mask(tested, prev_tested_population))
 
+    def update_vmmc_after_test(self, pop, time_step):
+        """
+        Update VMMC in individuals that tested HIV negative last time step.
+        """
+        if pop.circumcision.circ_after_test:
+            # select uncircumcised men tested last timestep
+            tested_uncirc_male_pop = pop.get_sub_pop([(col.SEX, op.eq, SexType.Male),
+                                                      (col.CIRCUMCISED, op.eq, False),
+                                                      (col.HIV_STATUS, op.eq, False),
+                                                      (col.LAST_TEST_DATE, op.eq, pop.date - time_step),
+                                                      (col.HARD_REACH, op.eq, False),
+                                                      (col.AGE, op.le, pop.circumcision.max_vmmc_age)])
+            # continue if eligible men are present this timestep
+            if len(tested_uncirc_male_pop) > 0:
+                # calculate post-test vmmc outcomes
+                r = rng.uniform(size=len(tested_uncirc_male_pop))
+                circumcision = r < pop.circumcision.prob_circ_after_test
+                # assign outcomes
+                pop.set_present_variable(col.CIRCUMCISED, circumcision, tested_uncirc_male_pop)
+                pop.set_present_variable(col.VMMC, circumcision, tested_uncirc_male_pop)
+
+    def update_vmmc_hiv_testing(self, pop, time_step):
+        """
+        Update HIV testing status after VMMC.
+        """
+        # those that just got circumcised and weren't tested last time step get tested now
+        just_tested = pop.get_sub_pop([(col.CIRCUMCISION_DATE, op.eq, pop.date),
+                                       (col.LAST_TEST_DATE, op.ne, pop.date - time_step)])
```

Comment:
This seems very specific and might be related to a 3-monthly time step. I suspect that if the timestep were one month then we would actually want to be checking for anyone who has not had a test in the last three months, but we should double check with the team.

Reply:
Looks like this entire function isn't doing what was intended by the SAS code, so we should remove it.